### ISSUE TITLE
bounding box queries over the dateline are not always returning results

### ISSUE THREAD

**Issue by Andrekra** [mapbox/supercluster#69]
When the bounding box covers bother sides of 0 (Quandrant 4 and 1 of a circle). for example ```index.getClusters([2, -10, -178, 10], 1)``` I expect it to return the points from 0 to 2 and -178 to -180.

I've wrote a test:
```
describe('group', () => {
  const createSpot = (horizonral, vertical) => {
    return {
      type: "Feature",
      geometry: {
        type: "Point",
        coordinates: [horizontal, vertical]
      }
    }
  }

  var index = supercluster({
    radius: 0.01,
    maxZoom: 1,
    extent: 512
  })

  it('should group spots near the edge', () => {
    index.load([
      createSpot(-179.989, 0),
      createSpot(-179.990, 0),
      createSpot(-179.991, 0),
      createSpot(-179.992, 0),
    ])

    const spots = index.getClusters([2, -10, -178, 10], 1)

    return expect(spots.length).toEqual(4)
  })
})
```
But it fails.
Am I missing something? Isn't the horizontal range -180 to 180 (vertical 90 to -90)? http://wiki.openstreetmap.org/wiki/Bounding_Box

**Comment by mourner** [comment]
Yeah, the bounding box values currently don't have the special treatment for the dateline. Want to work on a PR for this?

**Comment by Andrekra** [comment]
I'd love to help but my math regarding spherical coordinates is weak. Could you point me in the right direction? If I had something tangible I could work it out to a PR

**Comment by mourner** [comment]
I think we'll have to split any bboxes that go over the dateline into two queries — one on the right side (up to 180), and the other on the left (starting from -180)

**Comment by Andrekra** [comment]
Would you still be able to create 1 cluster from spots on either side of the dateline? Or would that create 2 clusters (1 from each bbox), that in theory could be right next to eachother.

**Comment by Andrekra** [comment]
@mourner I came across this extension for kdbush https://github.com/mourner/geokdbush. But it uses a point and a radius instead of 4 points to determine bbox. Would this be possible to integrate into supercluster? or are the concepts to far apart?

**Comment by mattlubner** [comment]
I'm also experiencing this issue, can I submit a PR to split it into two queries?

**Comment by mourner** [comment]
Should be fixed in #71 (thanks @mattlubner!). @Andrekra Does the PR look good to you?
